src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+#include <libhal-sensor/multi/bmp180.hpp>
+
+#include <array>
+
+#include <libhal-util/i2c.hpp>
+#include <libhal/error.hpp>
+
+using namespace std::literals;
+namespace hal::sensor {
+namespace {
+
+// functions for creating arrays containing the different register addresses in
+// their payloads
+auto device_id_register()
+{
+  return std::to_array<hal::byte>({ 0xD0 });
+}
+
+auto calibration_coefficients_register()
+{
+  return std::to_array<hal::byte>({ 0xAA });
+}
+
+auto control_measurement_register()
+{
+  return std::to_array<hal::byte>({ 0xF4 });
+}
+
+auto uncompensated_output_data_register()
+{
+  return std::to_array<hal::byte>({ 0xF6 });
+}
+
+auto soft_reset_register()
+{
+  return std::to_array<hal::byte>({ 0xE0 });
+}
+
+// struct to hold the computation variables
+struct computation_variables
+{
+  std::int32_t x1, x2, x3;
+  std::int32_t b3, b5, b6;
+  std::uint32_t b4, b7;
+  std::int32_t temperature, pressure;
+};
+
+// struct to hold temperature helper function results
+struct temperature_results
+{
+  hal::celsius temperature;
+  std::int32_t b5;
+};
+
+// helper function to check if sensor has finished its data conversion
+int conversion_status(hal::i2c* p_i2c, hal::byte p_address)
```

Comment:
```suggestion
bool conversion_status(hal::i2c* p_i2c, hal::byte p_address)
```
Consider returning a bool since this function only returns 1 or 0.